File: can-place-flowers/can-place-flowers.js
/**
 * @param {number[]} flowerbed
 * @param {number} n
 * @return {boolean}
 */
var canPlaceFlowers = function(flowerbed, n) {
    
    if(n== 0)return true;
    
    for(let i=0; i< flowerbed.length; i++){
        
        if(flowerbed[i]==0 && flowerbed[i-1] != 1 && flowerbed[i+1] !=1){
              flowerbed[i] =1
            n--
        }
        
        if(n==0)return true
        
    }
    
    return false
};

File: can-place-flowers/can-place-flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        
       
       
        flowerbed = [0] + flowerbed + [0]
        print(len(flowerbed))
        
        for i in range(1,len(flowerbed)-1):
            
            if i == 1 and flowerbed[i-1] == 0 and flowerbed[i+1]==0 and flowerbed[i]==0 and n > 0:
                flowerbed[i] = 1
                n -=1
                print("n2",n)
            elif i == len(flowerbed)-1 and flowerbed[i-1] == 0 and flowerbed[i+1] == 0 and n > 0 :
                n -=1
                print("n1",n)
            elif flowerbed[i] == 0 and flowerbed[i-1]==0 and flowerbed[i+1]== 0 and n > 0:
                flowerbed[i] =1
                n -=1
                print("n2",n)
            
        print(n,flowerbed)
        
        
        return (True if n == 0 else False)
        